extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -120,14 +117,18 @@ export class AmpConsent extends AMP.BaseElement {
     user().assert(this.element.getAttribute('id'),
         'amp-consent should have an id');
 
-    const inlineConfig = this.getInlineConfig_();
+    const config = new ConsentConfig(this.element);
```

Comment:
Thanks for splitting out the config into seperate functions!

I still have some reservations about the order of this code though. I noticed we are casting Object `/** @type {!Object} */` on our configs, such as line 147 since we are setting our configs later conditionally, and have to type them as nullable.

Can I suggest we do this?

```
...Class property delcarations...

const config = new ConsentConfig(this.element);

/** @private {!JsonObject} */
this.consentConfig_ = config.getConsentConfig();

...Other Config Properties...

...Class property delcarations...
```

Reply:
I agree with that. However based on previous lessons learned it's not suggested to read attribute in `constructor`. Please refer to #10058.

Reply:
Ahhh makes sense. Awesome. Thanks for clarifying that! 😄